### PR TITLE
[SPARK-13590] [ML] [Doc] Document spark.ml LiR, LoR and AFTSurvivalRegression behavior difference 

### PR DESCRIPTION
**Pull request by yanboliang**
## What changes were proposed in this pull request?

When fitting `LinearRegressionModel`(by "l-bfgs" solver) and `LogisticRegressionModel` w/o intercept on dataset with constant nonzero column, spark.ml produce same model as R glmnet but different from LIBSVM.

When fitting `AFTSurvivalRegressionModel` w/o intercept on dataset with constant nonzero column, spark.ml produce different model compared with R survival::survreg.

We should output a warning message and clarify in document for this condition.
## How was this patch tested?

Document change, no unit test.

cc @mengxr

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
(This is a super minor but I think `cc @mengxr` should be removed because the PR description explains the PR itself and the names of reviewers might not be related with the PR itself. `@` will be removed by Python merge script when merging but `cc` and `mengxr` will still remain, eg. https://github.com/apache/spark/commit/4514aebd1e807a665c270bfdc3f1127b3a1da898. Also I remember I was told so by one of committers)

File: docs/ml-classification-regression.md [hunk 1]
```diff
@@ -62,6 +62,8 @@ For more background and more details about the implementation, refer to the docu
 
   > The current implementation of logistic regression in `spark.ml` only supports binary classes. Support for multiclass regression will be added in the future.
 
+  > When fitting LogisticRegressionModel without intercept on dataset with constant nonzero column, Spark ML produce same model as R glmnet but different from LIBSVM.
```
Comment (mengxr) [inline docs/ml-classification-regression.md]: We should be more explicit since we know the corresponding coefficients are zeros.

**Comment by mengxr** [comment]
@yanboliang Please rename "Spark ML" to "MLlib". "Spark ML" is not an official name of the component. Thanks!

**Comment by yanboliang** [comment]
ping @mengxr

**Comment by mengxr** [comment]
LGTM

**Pull request closed without merging**